### ISSUE TITLE
Kubernetes node e2e tests

### ISSUE THREAD

**Issue by resouer** [kubernetes/frakti#109]
Node e2e test (conformance) results of latest **frakti** on master branch + **hyperd 0.8.0.1** + **Kubernetes v160-beta.4**.

The test command is:
```
make test-e2e-node PARALLELISM=2 TEST_ARGS='--kubelet-flags="--container-runtime=remote --container-runtime-endpoint=/var/run/frakti.sock --feature-gates=AllAlpha=true,Accelerators=false"' FOCUS="\[Conformance\]"
```
The node e2e result is:

```
Summarizing 9 Failures:

[Fail] [k8s.io] EmptyDir volumes [It] should support (non-root,0644,tmpfs) [Conformance] [Volume] 
/root/go-project/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2197

[Fail] [k8s.io] EmptyDir volumes [It] should support (non-root,0777,default) [Conformance] [Volume] 
/root/go-project/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2197

[Fail] [k8s.io] Projected [It] should project all components that make up the projection API [Conformance] [Volume] [Projection] 
/root/go-project/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2197

[Fail] [k8s.io] EmptyDir volumes [It] should support (non-root,0644,default) [Conformance] [Volume] 
/root/go-project/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2197

[Fail] [k8s.io] EmptyDir volumes [It] should support (non-root,0666,default) [Conformance] [Volume] 
/root/go-project/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2197

[Fail] [k8s.io] KubeletManagedEtcHosts [It] should test kubelet managed /etc/hosts file [Conformance] 
/root/go-project/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/exec_util.go:107

[Fail] [k8s.io] EmptyDir volumes [It] should support (non-root,0777,tmpfs) [Conformance] [Volume] 
/root/go-project/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2197

[Fail] [k8s.io] Kubelet when scheduling a read only busybox container [It] it should not write to root filesystem [Conformance] 
/root/go-project/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e_node/kubelet_test.go:151

[Fail] [k8s.io] EmptyDir volumes [It] should support (non-root,0666,tmpfs) [Conformance] [Volume] 
/root/go-project/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2197

Ran 121 of 211 Specs in 1987.027 seconds
FAIL! -- 112 Passed | 9 Failed | 0 Pending | 90 Skipped 
```
In the 9 failed issues:
6 issues related mound volume with specified mode, which seems buggy in hyperd.
1 issue is know issue of hyperd https://github.com/hyperhq/hyperd/issues/564
2 issues still have no clear root cause found

cc @feiskyer

**Comment by feiskyer** [comment]
Related with https://github.com/hyperhq/hyperd/issues/572.

**Comment by Crazykev** [comment]
latest node e2e result:
```
Summarizing 5 Failures:

[Fail] [k8s.io] HostPath [It] should give a volume the correct mode [Conformance] [Volume] 
/root/go-project/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2183

[Fail] [k8s.io] Projected [It] should project all components that make up the projection API [Conformance] [Volume] [Projection] 
/root/go-project/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2183

[Fail] [k8s.io] KubeletManagedEtcHosts [It] should test kubelet managed /etc/hosts file [Conformance] 
/root/go-project/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/exec_util.go:107

[Fail] [k8s.io] Networking [k8s.io] Granular Checks: Pods [It] should function for node-pod communication: udp [Conformance] 
/root/go-project/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/networking_utils.go:550

[Fail] [k8s.io] Kubelet when scheduling a read only busybox container [It] it should not write to root filesystem [Conformance] 
/root/go-project/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e_node/kubelet_test.go:151

Ran 121 of 211 Specs in 2298.700 seconds
FAIL! -- 116 Passed | 5 Failed | 0 Pending | 90 Skipped 
```

**Comment by feiskyer** [comment]
With https://github.com/kubernetes/kubernetes/pull/40804 and latest hyperd, there are still 3 failed test cases:

```
test-e2e-node PARALLELISM=2 CONTAINER_RUNTIME_ENDPOINT=/var/run/frakti.sock RUNTIME=remote FOCUS="\[Conformance\]"

Summarizing 3 Failures:

[Fail] [k8s.io] Projected [It] should project all components that make up the projection API [Conformance] [Volume] [Projection]
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2184

[Fail] [k8s.io] KubeletManagedEtcHosts [It] should test kubelet managed /etc/hosts file [Conformance]
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/exec_util.go:107

[Fail] [k8s.io] Kubelet when scheduling a read only busybox container [It] it should not write to root filesystem [Conformance]
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e_node/kubelet_test.go:151

Ran 121 of 213 Specs in 2156.379 seconds
FAIL! -- 118 Passed | 3 Failed | 0 Pending | 92 Skipped
```

**Comment by feiskyer** [comment]
Filed https://github.com/hyperhq/hyperd/issues/577 for `should project all components that make up the projection API` test.

**Comment by Crazykev** [comment]
```
$ test-e2e-node PARALLELISM=2 CONTAINER_RUNTIME_ENDPOINT=/var/run/frakti.sock RUNTIME=remote FOCUS="\[Conformance\]"

Summarizing 7 Failures:

[Fail] [k8s.io] Kubelet when scheduling a read only busybox container [It] it should not write to root filesystem [Conformance] 
/root/go-project/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e_node/kubelet_test.go:151

[Fail] [k8s.io] Networking [k8s.io] Granular Checks: Pods [It] should function for intra-pod communication: http [Conformance] 
/root/go-project/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/networking_utils.go:433

[Fail] [k8s.io] Networking [k8s.io] Granular Checks: Pods [It] should function for node-pod communication: http [Conformance] 
/root/go-project/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/networking_utils.go:433

[Fail] [k8s.io] KubeletManagedEtcHosts [It] should test kubelet managed /etc/hosts file [Conformance] 
/root/go-project/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/pods.go:83

[Fail] [k8s.io] HostPath [It] should give a volume the correct mode [Conformance] [Volume] 
/root/go-project/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2202

[Fail] [k8s.io] Networking [k8s.io] Granular Checks: Pods [It] should function for node-pod communication: udp [Conformance] 
/root/go-project/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/networking_utils.go:433

[Fail] [k8s.io] Networking [k8s.io] Granular Checks: Pods [It] should function for intra-pod communication: udp [Conformance] 
/root/go-project/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/networking_utils.go:433

Ran 121 of 215 Specs in 2863.269 seconds
FAIL! -- 114 Passed | 7 Failed | 0 Pending | 94 Skipped 
```

Four more network related issues. 
BTW, current node-e2e test script have some issue, could fix manually before https://github.com/kubernetes/kubernetes/pull/45707 merged.

**Comment by Crazykev** [comment]
I run node e2e test in my local environment twice, this is the best result:
```
Summarizing 2 Failures:

[Fail] [k8s.io] KubeletManagedEtcHosts [It] should test kubelet managed /etc/hosts file [Conformance] 
/root/go-project/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/exec_util.go:107

[Fail] [k8s.io] Kubelet when scheduling a read only busybox container [It] it should not write to root filesystem [Conformance] 
/root/go-project/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e_node/kubelet_test.go:151

Ran 121 of 217 Specs in 2386.048 seconds
FAIL! -- 119 Passed | 2 Failed | 0 Pending | 96 Skipped 
```
The first one will be fixed by #145 , so ideally, we have only one case fail.

**Comment by feiskyer** [comment]
@Crazykev Whith version of kubernetes are you running? kubernetes master?

**Comment by Crazykev** [comment]
@feiskyer Yep, a brand new kubernetes master with 
```
commit 23f0fe8632bea74e449617fc95b984155bfc768e
Merge: 6f4e0b6 56803ec
Author: Kubernetes Submit Queue <k8s-merge-robot@users.noreply.github.com>
Date:   Tue May 16 21:28:02 2017 -0700
```

**Comment by feiskyer** [comment]
```
Summarizing 1 Failure:

[Fail] [k8s.io] Kubelet when scheduling a read only busybox container [It] it should not write to root filesystem [Conformance]
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e_node/kubelet_test.go:151

Ran 121 of 223 Specs in 1965.306 seconds
FAIL! -- 120 Passed | 1 Failed | 0 Pending | 102 Skipped
```

Only one conformance tests failed with latest master now, which should be fixed after #161.

**Comment by feiskyer** [comment]
All conformance tests passed now. Closing it.
